source/standards/performance-testing.html.md.erb

```diff
@@ -0,0 +1,99 @@
+
+### Plan test scenarios
+
+A scenario is a hypothetical event when your service will be under heavy load. For example, every year the [G-Cloud framework on the Digital Marketplace](https://www.gov.uk/guidance/the-g-cloud-framework-on-the-digital-marketplace) opens for applications from suppliers for several weeks. This activity increases load on the [Digital Marketplace](https://www.digitalmarketplace.service.gov.uk/) up to the deadline.
+
+When most of its application infrastructure moved to the [GOV.UK PaaS](https://www.cloud.service.gov.uk/) the Digital Marketplace service needed to retest the service. Its developers and performance analysts queried Google Analytics for popular journeys, modelling normal and peak capacity needs for the previous year’s G-Cloud framework application period.
+
+This helped the team calculate figures for the largest number of concurrent user sessions and Requests Per Second (RPS) Digital Marketplace was able to support. This made sure the new platform could still cope with its expected network traffic.
+
+### Inform people
+
+Inform people affected by your test. For example, tell hosting providers or developers planning to use the test environment of your test’s date, time and duration. Make sure everyone involved in the test agrees to its conditions.
+
+### Build your test
+
+You do not need to fully hand code your tests because most load testing tools like [Gatling](https://gatling.io/), include scenario recorders. Some test scenarios may still need manual preparation, for example, providing lists of test users or specific search terms.
+
+Gatling is a popular load testing tool used by many teams at GDS including [GOV.UK Pay](https://www.payments.service.gov.uk/), [GOV.UK Notify](https://www.notifications.service.gov.uk/) and [Digital Marketplace](https://www.digitalmarketplace.service.gov.uk/).
+
+## Configure your test environment
+
+Test environments must resemble your production environment as far as possible so tests are accurate. Isolate test environments to stop them affecting your production environment.
+
+Consider how to [monitor service performance](https://gds-way.cloudapps.digital/standards/monitoring.html) during the load test. Your service will likely already have dashboards and graphs set up for common metrics such as CPU and memory utilization. Decide if your current metrics are sufficient or if you need to turn on additional logging or configure profiling traces.
+
+Perform practice runs of your tests to make sure they work as expected. For example, you may encounter tooling issues related to [Cross-Site Request Forgery (CSRF)](https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)) in your test environment.
+
+## Run your tests
+
+Start with a low level of concurrent users and increase them to your anticipated maximum. If there are no problems, increase the amount of traffic until you reach a breaking point in your system.
```

Comment:
It might be worth mentioning something about planning for a certain buffer above your anticipated maximum. Yu might decided for example that you need to be able to support 1.5x your expected max, as a buffer.

Reply:
Good point but think this sits more in the capacity planning realm which I've not really covered in this guidance. We should revisit though if we don't think the service manual page on it covers enough ground